server/application/application.go

```diff
@@ -234,9 +236,47 @@ func (s *Server) Update(ctx context.Context, q *ApplicationUpdateRequest) (*appv
 	return s.appclientset.ArgoprojV1alpha1().Applications(s.ns).Update(a)
 }
 
-// UpdateSpec updates an application spec
+// UpdateSpec updates an application spec and filters out any invalid parameter overrides
 func (s *Server) UpdateSpec(ctx context.Context, q *ApplicationUpdateSpecRequest) (*appv1.ApplicationSpec, error) {
 	a, err := s.appclientset.ArgoprojV1alpha1().Applications(s.ns).Get(*q.Name, metav1.GetOptions{})
+	validAppSet := make(map[string]map[string]bool)
```

Comment:
1. The error check that used to come immediately after the Applications().Get() request, now comes after this large code block. So its possible to get a nil pointer exception while you are attempting to validate the params (a might be nil).

2. I think this validation logic is large enough that it should be put into a separate function to keep the UpdateSpec more concise.

Reply:
1. ah good point will reorder. 
2. ok, makes sense will create a helper function tomorrow.

thanks for the feedback!